src/main/ruby/core/posix.rb

```diff
@@ -27,37 +35,97 @@ def self.to_nfi_type(type)
+        string_args << i if arg_type == :string
+      }
+      string_args.freeze
+
+      define_singleton_method(method_name) { |*args|
+        string_args.each do |i|
+          str = args.fetch(i)
+          if str.encoding == Encoding::BINARY
+            str = str.dup.force_encoding(FS_ENCODING)
+          else
+            str = str.encode(FS_ENCODING)
+          end
+          args[i] = str
+        end
+
+        bound_func.call(*args)
+      }
+    else
+      define_singleton_method(method_name) { |*|
+        raise NotImplementedError, "#{native_name} is not available"
+      }
+      Truffle.invoke_primitive :method_unimplement, method(method_name)
+    end
   end
 
+  # Filesystem-related
   attach_function :access, [:string, :int], :int
+  attach_function :chdir, [:string], :int
   attach_function :chmod, [:string, :mode_t], :int
   attach_function :chown, [:string, :uid_t, :gid_t], :int
   attach_function :dup, [:int], :int
   attach_function :dup2, [:int, :int], :int
   attach_function :fchmod, [:int, :mode_t], :int
   attach_function :fchown, [:int, :uid_t, :gid_t], :int
   attach_function :fsync, [:int], :int
+  attach_function :lchmod, [:string, :mode_t], :int
+  attach_function :link, [:string, :string], :int
+  attach_function :mkdir, [:string, :mode_t], :int
   attach_function :mkfifo, [:string, :mode_t], :int
+  attach_function :readlink, [:string, :pointer, :size_t], :ssize_t
+  attach_function :rmdir, [:string], :int
+  attach_function :umask, [:mode_t], :mode_t
+  attach_function :unlink, [:string], :int
+  attach_function :utimes, [:string, :pointer], :int
 
+  # Process-related
   attach_function :getegid, [], :gid_t
-  attach_function :geteuid, [], :uid_t
   attach_function :getgid, [], :gid_t
+  attach_function :setresgid, [:gid_t, :gid_t, :gid_t], :int
+  attach_function :setregid, [:gid_t, :gid_t], :int
+  attach_function :setegid, [:uid_t], :int
+  attach_function :setgid, [:gid_t], :int
+
+  attach_function :geteuid, [], :uid_t
   attach_function :getuid, [], :uid_t
+  attach_function :setresuid, [:uid_t, :uid_t, :uid_t], :int
+  attach_function :setreuid, [:uid_t, :uid_t], :int
+  attach_function :setruid, [:uid_t], :int
   attach_function :seteuid, [:uid_t], :int
-  attach_function :setgid, [:gid_t], :int
   attach_function :setuid, [:uid_t], :int
 
+  attach_function :getpid, [], :pid_t
+  attach_function :getppid, [], :pid_t
+  attach_function :kill, [:pid_t, :int], :int
+  attach_function :getpgrp, [], :pid_t
+  attach_function :getpgid, [:pid_t], :pid_t
+  attach_function :setpgid, [:pid_t, :pid_t], :int
+  attach_function :setsid, [], :pid_t
+
+  attach_function :getgroups, [:int, [:gid_t]], :int
+
   attach_function :getrlimit, [:int, :pointer], :int
   attach_function :setrlimit, [:int, :pointer], :int
 
+  attach_function :getpriority, [:int, :id_t], :int
+  attach_function :setpriority, [:int, :id_t, :int], :int
+
+  # ENV-related
   attach_function :getenv_native, :getenv, [:string], :string
```

Comment:
`attach_function :getenv, [:string], :string, as: :getenv_native` would be more readable than 
`attach_function :getenv_native, :getenv, [:string], :string`

Reply:
This is gone anyway in a later commit.

Reply:
Ah this line is indeed gone, I actually meant to change the signature of `attach_function` to make these calls more readable when it's aliased. It's a nice to have change next time somebody edits it.

Reply:
The signature with two names as first/second argument is how Ruby FFI does it: https://github.com/ffi/ffi/wiki/Basic-Usage
We don't have to stick to it but it seems simpler initially.
I agree `as:` is more readable, PR welcome.

I was thinking of passing a block do attach_function, but it just seems more complicated.